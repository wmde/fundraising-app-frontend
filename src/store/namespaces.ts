export const NS_PAYMENT = 'payment';
export const NS_ADDRESS = 'address';
