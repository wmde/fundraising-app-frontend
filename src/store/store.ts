import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import createPayment from './payment';

import {
	NS_PAYMENT
} from './namespaces';

Vue.use(Vuex);

export function createStore() {
	const storeBundle: StoreOptions<any> = {
		modules: {
			[NS_PAYMENT]: createPayment()
		},
		strict: process.env.NODE_ENV !== 'production',
	};

	return new Vuex.Store<any>(storeBundle);
}
