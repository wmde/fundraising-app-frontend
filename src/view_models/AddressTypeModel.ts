export enum AddressTypeModel {
    PERSON,
    COMPANY,
    ANON,
}
