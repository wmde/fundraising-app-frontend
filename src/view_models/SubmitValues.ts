export interface TrackingData {
    bannerImpressionCount: number,
    impressionCount: number
}
