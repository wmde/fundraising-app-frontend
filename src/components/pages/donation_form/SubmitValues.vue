<template>
	<span class="submit-values">

		<input type="hidden" name="zahlweise" :value="payment.type">
		<input type="hidden" name="periode" :value="payment.interval">
		<input type="hidden" name="betrag" :value="formattedAmount">
		<input type="hidden" name="iban" :value="bankdata.iban">
		<input type="hidden" name="bic" :value="bankdata.bic">

		<input type="hidden" name="addressType" :value="addressType">

		<input type="hidden" name="salutation" :value="address.salutation">
		<input type="hidden" name="title" :value="address.title">
		<input type="hidden" name="firstName" :value="address.firstName">
		<input type="hidden" name="lastName" :value="address.lastName">
		<input type="hidden" name="companyName" :value="address.companyName">
		<input type="hidden" name="street" :value="address.street">
		<input type="hidden" name="postcode" :value="address.postcode">
		<input type="hidden" name="city" :value="address.city">
		<input type="hidden" name="country" :value="address.country">
		<input type="hidden" name="email" :value="address.email">
		<input type="hidden" name="info" :value="newsletterOptIn">
		<input type="hidden" name="donationReceipt" :value="receiptOptIn">

		<input type="hidden" name="impCount" :value="trackingData.bannerImpressionCount">
		<input type="hidden" name="bImpCount" :value="trackingData.impressionCount">

	</span>
</template>

<script lang="ts">
import Vue from 'vue';
import { mapState } from 'vuex';
import { NS_ADDRESS, NS_BANKDATA, NS_PAYMENT } from '@/store/namespaces';
import { Payment } from '@/view_models/Payment';
import { AddressState } from '@/view_models/Address';
import { addressTypeName } from '@/view_models/AddressTypeModel';
import { BankAccount } from '@/view_models/BankAccount';

export default Vue.extend( {
	name: 'SubmitValues',
	props: [
		'trackingData',
	],
	computed: {
		...mapState( NS_PAYMENT, {
			payment: state => ( state as Payment ).values,
			formattedAmount: state => {
				// endpoint expects a German-formatted decimal number
				const strAmount = String( ( state as Payment ).values.amount );
				return [
					strAmount.slice( 0, -2 ),
					',',
					strAmount.slice( -2 ),
				].join( '' );
			},
		} ),
		...mapState( NS_ADDRESS, {
			address: state => ( state as AddressState ).values,
			addressType: state => {
				return addressTypeName( ( state as AddressState ).addressType );
			},
			newsletterOptIn: state => ( state as AddressState ).newsletterOptIn ? '1' : '',
			receiptOptIn: state => ( state as AddressState ).receiptOptOut ? '0' : '1',
		} ),
		...mapState( NS_BANKDATA, {
			bankdata: state => ( state as BankAccount ).values,
		} ),
	},
} );
</script>
