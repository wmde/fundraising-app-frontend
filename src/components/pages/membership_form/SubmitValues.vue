<template>
	<span class="submit-values">

		<input type="hidden" name="membership_type" :value="membershipType">

		<input type="hidden" name="payment_type" value="BEZ">
		<input type="hidden" name="membership_fee_interval" :value="fee.interval">
		<input type="hidden" name="membership_fee" :value="formattedAmount">
		<input type="hidden" name="iban" :value="bankdata.iban">
		<input type="hidden" name="bic" :value="bankdata.bic">

		<input type="hidden" name="adresstyp" :value="addressType">

		<input type="hidden" name="anrede" :value="address.salutation">
		<input type="hidden" name="titel" :value="address.title">
		<input type="hidden" name="vorname" :value="address.firstName">
		<input type="hidden" name="nachname" :value="address.lastName">
		<input type="hidden" name="firma" :value="address.companyName">
		<input type="hidden" name="strasse" :value="address.street">
		<input type="hidden" name="postcode" :value="address.postcode">
		<input type="hidden" name="ort" :value="address.city">
		<input type="hidden" name="country" :value="address.country">
		<input type="hidden" name="email" :value="address.email">
		<input type="hidden" name="donationReceipt" :value="receiptOptIn">
		<input type="hidden" name="dob" :value="address.date">

	</span>
</template>

<script lang="ts">
import Vue from 'vue';
import { mapState } from 'vuex';
import { NS_BANKDATA, NS_MEMBERSHIP_ADDRESS, NS_MEMBERSHIP_FEE } from '@/store/namespaces';
import { Payment } from '@/view_models/Payment';
import { MembershipAddressState } from '@/view_models/Address';
import { addressTypeName } from '@/view_models/AddressTypeModel';
import { BankAccount } from '@/view_models/BankAccount';
import { membershipTypeName } from '@/view_models/MembershipTypeModel';

export default Vue.extend( {
	name: 'SubmitValues',
	computed: {
		...mapState( NS_MEMBERSHIP_FEE, {
			fee: state => ( state as Payment ).values,
			formattedAmount: state => {
				// endpoint expects a German-formatted decimal number
				const strAmount = String( ( state as Payment ).values.fee );
				return [
					strAmount.slice( 0, -2 ),
					',',
					strAmount.slice( -2 ),
				].join( '' );
			},
		} ),
		...mapState( NS_MEMBERSHIP_ADDRESS, {
			address: state => ( state as MembershipAddressState ).values,
			addressType: state => {
				return addressTypeName( ( state as MembershipAddressState ).addressType );
			},
			receiptOptIn: state => ( state as MembershipAddressState ).receiptOptOut ? '0' : '1',
			membershipType: state => membershipTypeName( ( state as MembershipAddressState ).membershipType ),
		} ),
		...mapState( NS_BANKDATA, {
			bankdata: state => ( state as BankAccount ).values,
		} ),
	},
} );
</script>
