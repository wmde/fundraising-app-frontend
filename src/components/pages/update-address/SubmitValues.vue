<template>
	<span class="submit-values">
		<input type="hidden" name="addressType" :value="addressType">

		<input type="hidden" name="salutation" :value="address.salutation">
		<input type="hidden" name="title" :value="address.title">
		<input type="hidden" name="firstName" :value="address.firstName">
		<input type="hidden" name="lastName" :value="address.lastName">
		<input type="hidden" name="companyName" :value="address.companyName">
		<input type="hidden" name="street" :value="address.street">
		<input type="hidden" name="postcode" :value="address.postcode">
		<input type="hidden" name="city" :value="address.city">
		<input type="hidden" name="country" :value="address.country">
		<input type="hidden" name="donationReceipt" :value="receiptOptIn">
	</span>
</template>

<script lang="ts">
import Vue from 'vue';
import { mapState } from 'vuex';
import { NS_ADDRESS, NS_BANKDATA, NS_PAYMENT } from '@/store/namespaces';
import { Payment } from '@/view_models/Payment';
import { AddressState } from '@/view_models/Address';
import { addressTypeName } from '@/view_models/AddressTypeModel';
import { BankAccount } from '@/view_models/BankAccount';

export default Vue.extend( {
	name: 'SubmitValues',
	props: [
		'trackingData',
	],
	computed: {
		...mapState( NS_ADDRESS, {
			address: ( state: AddressState ) => state.values,
			addressType: ( state: AddressState ) => {
				return addressTypeName( state.addressType );
			},
			receiptOptIn: ( state: AddressState ) => state.receiptOptOut ? '0' : '1',
		} ),
	},
} );
</script>
