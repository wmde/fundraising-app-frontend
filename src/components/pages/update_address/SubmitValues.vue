<template>
	<span class="submit-values">
		<input type="hidden" name="addressType" :value="addressType">

		<input type="hidden" name="salutation" :value="address.salutation">
		<input type="hidden" name="title" :value="address.title">
		<input type="hidden" name="firstName" :value="address.firstName">
		<input type="hidden" name="lastName" :value="address.lastName">
		<input type="hidden" name="company" :value="address.companyName">
		<input type="hidden" name="street" :value="address.street">
		<input type="hidden" name="postcode" :value="address.postcode">
		<input type="hidden" name="city" :value="address.city">
		<input type="hidden" name="country" :value="address.country">
		<input type="hidden" name="receiptOptOut" :value="receiptOptOut">
	</span>
</template>

<script lang="ts">
import Vue from 'vue';
import { mapState } from 'vuex';
import { NS_ADDRESS, NS_BANKDATA, NS_PAYMENT } from '@/store/namespaces';
import { Payment } from '@/view_models/Payment';
import { AddressState } from '@/view_models/Address';
import { addressTypeName } from '@/view_models/AddressTypeModel';
import { BankAccount } from '@/view_models/BankAccount';

export default Vue.extend( {
	name: 'SubmitValues',
	props: [
		'trackingData',
	],
	computed: {
		...mapState( NS_ADDRESS, {
			address: state => ( state as AddressState ).values,
			addressType: state => {
				return addressTypeName( ( state as AddressState ).addressType );
			},
			receiptOptOut: state => ( state as AddressState ).receiptOptOut ? '1' : '',
		} ),
	},
} );
</script>
