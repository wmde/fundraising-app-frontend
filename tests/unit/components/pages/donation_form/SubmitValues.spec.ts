import { createLocalVue, mount } from '@vue/test-utils';
import Vuex from 'vuex';
import SubmitValues from '@/components/pages/donation_form/SubmitValues.vue';
import { NS_BANKDATA, NS_ADDRESS, NS_PAYMENT } from '@/store/namespaces';
import { AddressTypeModel, addressTypeName } from '@/view_models/AddressTypeModel';

const localVue = createLocalVue();
localVue.use( Vuex );

const getWrapper = () => {
	return mount( SubmitValues, {
		localVue,
		propsData: {
			trackingData: {
				bannerImpressionCount: 1,
				impressionCount: 5,
			},
			campaignValues: {
				campaign: 'nicholas',
				keyword: 'cage',
			},
		},
		store: new Vuex.Store( {
			modules: {
				[ NS_ADDRESS ]: {
					namespaced: true,
					state: {
						receiptOptOut: false,
						newsletterOptIn: true,
						addressType: AddressTypeModel.PERSON,
						values: {
							firstName: 'Victor',
							lastName: 'van Doom',
							salutation: 'Herr',
							title: 'Dr.',
							street: 'Untere StraÃŸe 5',
							postcode: '08114',
							city: 'Haasenstadt',
							country: 'DE',
							email: 'doom@untergang.biz',
						},
					},

				},
				[ NS_BANKDATA ]: {
					namespaced: true,
					state: {
						values: {
							iban: 'DE12500105170648489890',
							bic: 'INGDDEFFXXX',
						},
					},
				},
				[ NS_PAYMENT ]: {
					namespaced: true,
					state: {
						values: {
							amount: '2349',
							interval: '3',
							type: 'BEZ',
						},
					},
				},
			},
		} ),
	} );
};

describe( 'SubmitValues.vue', () => {
	it( 'renders input fields', () => {
		const wrapper = getWrapper();
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'renders the amount as an integer', () => {
		const wrapper = getWrapper();
		expect( ( wrapper.find( 'input[name=amount]' ).element as HTMLInputElement ).value ).toBe( '2349' );
	} );

	it( 'renders the address type as string', () => {
		const wrapper = getWrapper();
		expect( ( wrapper.find( 'input[name=addressType]' ).element as HTMLInputElement ).value ).toBe( addressTypeName( AddressTypeModel.PERSON ) );
	} );

	it( 'sends tracking values', () => {
		const wrapper = getWrapper();

		expect( ( wrapper.find( 'input[name=bImpCount]' ).element as HTMLInputElement ).value ).toBe( '1' );
		expect( ( wrapper.find( 'input[name=impCount]' ).element as HTMLInputElement ).value ).toBe( '5' );
	} );

	it( 'sends campaign values', () => {
		const wrapper = getWrapper();

		expect( ( wrapper.find( 'input[name=piwik_campaign]' ).element as HTMLInputElement ).value ).toBe( 'nicholas' );
		expect( ( wrapper.find( 'input[name=piwik_kwd]' ).element as HTMLInputElement ).value ).toBe( 'cage' );
	} );
} );
