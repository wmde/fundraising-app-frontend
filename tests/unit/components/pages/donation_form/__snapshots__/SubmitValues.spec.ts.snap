// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address.vue renders input fields 1`] = `
<span
  class="submit-values"
>
  <input
    name="zahlweise"
    type="hidden"
    value="BEZ"
  />
   
  <input
    name="periode"
    type="hidden"
    value="3"
  />
   
  <input
    name="betrag"
    type="hidden"
    value="23,49"
  />
   
  <input
    name="iban"
    type="hidden"
    value="DE12500105170648489890"
  />
   
  <input
    name="bic"
    type="hidden"
    value="INGDDEFFXXX"
  />
   
  <input
    name="addressType"
    type="hidden"
    value="person"
  />
   
  <input
    name="salutation"
    type="hidden"
    value="Herr"
  />
   
  <input
    name="title"
    type="hidden"
    value="Dr."
  />
   
  <input
    name="firstName"
    type="hidden"
    value="Victor"
  />
   
  <input
    name="lastName"
    type="hidden"
    value="van Doom"
  />
   
  <input
    name="companyName"
    type="hidden"
  />
   
  <input
    name="street"
    type="hidden"
    value="Untere StraÃŸe 5"
  />
   
  <input
    name="postcode"
    type="hidden"
    value="08114"
  />
   
  <input
    name="city"
    type="hidden"
    value="Haasenstadt"
  />
   
  <input
    name="country"
    type="hidden"
    value="DE"
  />
   
  <input
    name="email"
    type="hidden"
    value="doom@untergang.biz"
  />
   
  <input
    name="info"
    type="hidden"
    value="1"
  />
   
  <input
    name="donationReceipt"
    type="hidden"
    value="1"
  />
   
  <input
    name="impCount"
    type="hidden"
    value="1"
  />
   
  <input
    name="bImpCount"
    type="hidden"
    value="5"
  />
</span>
`;
